docs/references/java_project/index.md

```diff
@@ -1,17 +1,96 @@
+# Overview 
+
+
+### Templates
+
+All the templates generate the base project structure along with some additional files and directories specific to 
```

Comment:
Might be good to link back to the base project structure description as well - similarly for the python overview